libs/platform/src/lib/components/input-group/addon/addon.component.ts

```diff
@@ -0,0 +1,133 @@
+import {
+    Component,
+    ChangeDetectionStrategy,
+    OnInit,
+    ElementRef,
+    AfterContentInit,
+    ContentChildren,
+    QueryList,
+    Renderer2
+} from '@angular/core';
+import { startWith } from 'rxjs/operators';
+
+import { ContentDensity } from '../../form/form-control';
+import { ButtonComponent } from '../../button/button.component';
+
+const CSS_CLASS_NAME = {
+    addon: 'fd-input-group__addon',
+    addonButton: 'fd-input-group__addon--button',
+    addonCompact: 'fd-input-group__addon--compact',
+    innerButton: 'fd-input-group__button'
+} as const;
+
+/**
+ * Fundamental input group addon component
+ *
+ * This component is created to be used inside `<fdp-input-group>`
+ *
+ *
+ * Input group addon can contain a plain text:
+ * ```
+ * <fdp-input-group-addon>$</fdp-input-group-addon>
+ * ```
+ * Icon:
+ * ```
+ * <fdp-input-group-addon><i class="sap-icon--employee"></i></fdp-input-group-addon>
+ * ```
+ * Button component:
+ * ```
+ * <fdp-input-group-addon><fdp-button>Action</fdp-button></fdp-input-group-addon>
+ * ```
+ *
+ */
+@Component({
+    selector: 'fdp-input-group-addon',
+    templateUrl: './addon.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InputGroupAddonComponent implements OnInit, AfterContentInit {
+    /**
+     * content Density of element: 'cozy' | 'compact'
+     */
+    set contentDensity(contentDensity: ContentDensity) {
+        if (contentDensity === this._contentDensity) {
+            return;
+        }
+
+        this._contentDensity = contentDensity;
+
+        if (contentDensity === 'compact') {
+            this._addClassNameToHostElement(CSS_CLASS_NAME.addonCompact);
+        } else {
+            this._removeClassNameFromHostElement(CSS_CLASS_NAME.addonCompact);
+        }
+
+        this._setFdpButtonsOptions();
+    }
+
+    /**@hidden */
+    @ContentChildren(ButtonComponent)
+    private _fdpButtons: QueryList<ButtonComponent>;
+
+    /** @hidden */
+    private _contentDensity: ContentDensity;
+
+    /**@hidden */
+    constructor(private _elementRef: ElementRef<HTMLElement>, private _renderer: Renderer2) {}
+
+    /**@hidden */
+    ngOnInit(): void {
+        this._addClassNameToHostElement(CSS_CLASS_NAME.addon);
+    }
+
+    /**@hidden */
+    ngAfterContentInit(): void {
+        this._fdpButtons.changes
```

Comment:
Never unsubscribed. I'd recommend to move this to separate method, to keep the life-cycle hooks clean.

Reply:
We can skip it as angular completes subscription automatically.
Moved subscription to a separate method.